### PR TITLE
Add the Medical Safety Alert specialist document format

### PR DESCRIPTION
**Pull request by issyl0**
- Add a schema for it, metadata form fields, a model and its
  validations, and model specs.

https://trello.com/c/UFpwZWOS/537-msa

### CODE REVIEW

File: spec/models/medical_safety_alert_spec.rb [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      "description" => "This is the summary of example Medical Safety Alert #{n}",
+      "format" => "specialist_document",
+      "publishing_app" => "specialist-publisher",
+      "rendering_app" => "specialist-frontend",
+      "locale" => "en",
+      "phase" => "live",
+      "public_updated_at" => "2015-11-16T11:53:30",
+      "publication_state" => "draft",
+      "details" => {
+        "body" => "## Header" + ("\r\n\r\nThis is the long body of an example Medical Safety Alert" * 10),
```
Comment (jamiecobbett) [inline spec/models/medical_safety_alert_spec.rb]: Should this really be Govspeak?

File: spec/models/medical_safety_alert_spec.rb [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      "base_path" => "/government/organisations/medicines-and-healthcare-products-regulatory-agency",
+      "content_id" => "240f72bd-9a4d-4f39-94d9-77235cadde8e",
+      "title" => "Medicines and Healthcare products Regulatory Agency",
+      "format" => "placeholder_organisation",
+      "need_ids" => [],
+      "locale" => "en",
+      "updated_at" => "2015-12-01T15:51:43.521Z",
+      "public_updated_at" => "2015-06-29T14:33:29.000+00:00",
+      "phase" => "live",
+      "analytics_identifier" => "EA63",
```
Comment (jamiecobbett) [inline spec/models/medical_safety_alert_spec.rb]: Is this field needed? The original purpose was to use it for organisations and world locations.

File: app/views/metadata_fields/_medical_safety_alerts.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<%= f.label :alert_type %>
+<%= f.select :alert_type, f.object.facet_options(:alert_type), {include_blank: true}, { class: 'form-control' } %>
```
Comment (jamiecobbett) [inline app/views/metadata_fields/_medical_safety_alerts.html.erb]: Our [styleguide says](https://github.com/alphagov/styleguides/blob/master/ruby.md#general) we should put spaces inside `{` and `}`.

File: spec/models/medical_safety_alert_spec.rb (hunk 1 above)
Comment (tommyp) [inline spec/models/medical_safety_alert_spec.rb]: This is just copying what is sent from master. As part of migration work, master will need to send Govspeak and rendered Govspeak, in which case these tests will need to be updated.

File: spec/models/medical_safety_alert_spec.rb (hunk 2 above)
Comment (tommyp) [inline spec/models/medical_safety_alert_spec.rb]: This was copied from the existing response. Only the `base_path`, `content_id` and `title` are required.

Comment (jamiecobbett) [inline spec/models/medical_safety_alert_spec.rb]: Is it realistic then? Do Specialist Documents ever use that field?

Comment (tommyp) [inline spec/models/medical_safety_alert_spec.rb]: It's realistic in that it is the response, but it's not used.

File: spec/models/medical_safety_alert_spec.rb (hunk 1 above)
Comment (tommyp) [inline spec/models/medical_safety_alert_spec.rb]: Also of note, this does *need* to be Govspeak right now, as this is what the User is editing every time they edit the document.

File: app/views/metadata_fields/_medical_safety_alerts.html.erb (hunk 3 above)
Comment (issyl0) [inline app/views/metadata_fields/_medical_safety_alerts.html.erb]: Fixed.

**Comment by boffbowsh** [comment]
See the changes in https://github.com/alphagov/specialist-publisher/pull/655 for the style of these specs

**Comment by issyl0** [comment]
Thanks @tommyp, @boffbowsh. I'll get round to updating the specs tomorrow, promise. :sunny:

**Comment by issyl0** [comment]
@tommyp @boffbowsh I've changed the style of the specs as per #656.

File: spec/models/medical_safety_alert_spec.rb [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    publishing_api_has_fields_for_format('specialist_document', medical_safety_alerts, fields)
+
+    medical_safety_alerts.each do |medical_safety_alert|
+      publishing_api_has_item(medical_safety_alert)
+    end
+
+    Timecop.freeze(Time.parse("2015-12-18 10:12:26 UTC"))
+  end
+
+  describe "#all" do
```
Comment (boffbowsh) [inline spec/models/medical_safety_alert_spec.rb]: Nitpicky, but these should be `describe ".all"` and `describe ".find"` as they're class methods.

Comment (issyl0) [inline spec/models/medical_safety_alert_spec.rb]: Ugh. I'll make a note to update where I copied and pasted them from then as well as here and presumably https://github.com/alphagov/specialist-publisher/pull/656/files#diff-87a2b2747a4f709e7ef875915e7fcee5R106 too.

Comment (issyl0) [inline spec/models/medical_safety_alert_spec.rb]: Thanks. Done.

**Comment by Rosa-Fox** [comment]
@issyl0 Teeny tiny thing which is that `".save!"` on 135 of `medial_safety_alert_spec.rb` should be `#save!`

**Comment by issyl0** [comment]
@Rosa-Fox Oh! Thanks. Fixed.

**Pull request merged**